### ISSUE TITLE
No source artifacts are imported with SBT 1.3.0-RC2

### ISSUE THREAD

**Issue by gemelen** [scalameta/metals#831]
**Describe the bug**
In project managed with SBT 1.3.0-RC2 no sources are added to bloop configuration files (.bloop/<module>.json) for any dependency that is already present in local Ivy cache and project sources. 
Thus, "Go to definition" feature doesn't work for any symbol from missed artifacts. Message "no definitions found" is shown.
.bloop/<module>.json contains no artifact of "sources" classifier in project.resolution.modules

**To Reproduce**
1. Create any SBT project with SBT version set to 1.3.0-RC2 and some external dependencies
2. Make sure that there is no src jars for those dependencies in Ivy cache (move/remove cache dir content)
3. Open project in editor and auto-import it
4. Import succeeds, all Metals features work, except "Go to definition" feature for symbols from external dependency responses with "no definitions found" message.

**Expected behavior**
Sources for dependencies should be added to project definition and downloaded for definition resolution to work properly.

Tested only with setup below, but seems to be irrelevant to editors themselves and rather be integration with SBT (which switch to Coursier for artifact management) issue.
**Installation:**
- Operating system: Linux
- Editor: Vim (vim-lsc), Neovim (coc.nvim)
- Metals version: v0.7.0

**Additional context**
Workaround: downgrade SBT version to 1.2.8, re-import project, upgrade SBT version back to 1.3.0-RC2

**Search terms**
SBT 1.3.0-RC2
"no definition found"
"Coursier"

**Comment by tgodzik** [comment]
Thanks for reporting! Does this also happens if you add bloop plugin manually and run bloopInstall like defined in `https://scalacenter.github.io/bloop/setup#universal` ?

This is most likely an issue in Bloop itself, so we would want to fix it there.

**Comment by gemelen** [comment]
@tgodzik yes, I've just tried this instruction for manual bloop setup and `sbt bloopInstall` generated same project definition with no source artifacts.

**Comment by tgodzik** [comment]
@jvican should we move this issue to Bloop and try to fix it for RC2 or wait until full release?

**Comment by jvican** [comment]
My impression is that this s a bug in sbt so we should report it there, bloop is not doing anything special here, it's just using the public sbt commands to resolve source artifacts (see https://github.com/scalacenter/bloop/blob/765d745a9b87b061cb76127eea44e3902a857b5d/integrations/sbt-bloop/src/main/scala/bloop/integrations/sbt/SbtBloop.scala#L90-L95 and https://github.com/scalacenter/bloop/blob/765d745a9b87b061cb76127eea44e3902a857b5d/integrations/sbt-bloop/src/main/scala/bloop/integrations/sbt/SbtBloop.scala#L627-L632). If that is not working with sbt 1.3.0-RC2, then it means there's something wrong with the coursier integration in the upcoming sbt 1.3.0.

**Comment by gemelen** [comment]
Seems like it's known issue already, sbt/sbt/issues/4831
I'll check as soon as RC3 with the fix would be available.

**Comment by gemelen** [comment]
Yay 🥳. It works as supposed to be with SBT 1.3.0-RC3.
